pkg/controllers/dynakube/otlp/reconciler_test.go

```diff
@@ -0,0 +1,257 @@
+package otlp
+
+import (
+	"context"
+	"errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"testing"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/latest/dynakube"
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/latest/dynakube/otlpexporterconfiguration"
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/scheme/fake"
+	dtclient "github.com/Dynatrace/dynatrace-operator/pkg/clients/dynatrace"
+	"github.com/Dynatrace/dynatrace-operator/pkg/consts"
+	"github.com/Dynatrace/dynatrace-operator/pkg/otlp/exporterconfig"
+	dtwebhook "github.com/Dynatrace/dynatrace-operator/pkg/webhook/mutation/pod/mutator"
+	dtclientmock "github.com/Dynatrace/dynatrace-operator/test/mocks/pkg/clients/dynatrace"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	testDataIngestToken = "test-ingest-token"
+
+	testDynakube   = "test-dynakube"
+	testDynakube2  = "test-dynakube2"
+	testNamespace  = "test-namespace"
+	testNamespace2 = "test-namespace2"
+
+	testNamespaceSelectorLabel = "namespaceSelector"
+
+	testNamespaceDynatrace = "dynatrace"
+)
+
+// failingReader simulates failures on List operations
+type failingReader struct {
+	client.Reader
+	fail bool
+}
+
+func (f failingReader) List(ctx context.Context, list client.ObjectList, opts ...client.ListOption) error { //nolint:revive
+	if f.fail {
+		return errors.New("err")
+	}
+	return f.Reader.List(ctx, list, opts...)
+}
+
+func TestReconciler_Reconcile(t *testing.T) {
+	t.Run("reconcile OTLP exporter configuration", func(t *testing.T) {
+		dk := &dynakube.DynaKube{
```

Comment:
the tests all use very similar dynakubes. a constructor would reduce duplication